app/scripts/lib/fxa-client.js

```diff
@@ -87,6 +87,24 @@ define(function (require, exports, module) {
     };
   }
 
+  /**
+   * Create a delegate method to the fxa-js-client.
+   *
+   * @param {String} method to delegate to.
+   * @returns {Function}
+   */
+  function createClientDelegate(method) {
+    return function (...args) {
+      return this._getClient()
+        .then((client) => {
+          if (! client[method]) {
```

Comment:
Any mileage in tightening this up to ensure we're dealing with a function:

```js
if (typeof client[method] !== 'function') {
```

?